### PR TITLE
[WIN32SS] Create pathdraw.c

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
Create `win32ss/gdi/eng/pathdraw.c` and move `NtGdiEngStrokePath`, `NtGdiEngFillPath`, `NtGdiEngStrokeAndFillPath`, `EngStrokePath`, `EngFillPath` and `EngStrokeAndFillPath` functions to this file.
JIRA issue: [CORE-16243](https://jira.reactos.org/browse/CORE-16243)

### CODE REVIEW

**Comment by jimtabor** [comment]
It's good to go. I've made changes in the stub file and see no problem.

File: win32ss/gdi/eng/pathdraw.c [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+    _SEH2_TRY
+    {
+        ProbeForRead(pptlBrushOrg, sizeof(*pptlBrushOrg), 1);
+        ptlBrushOrg = *pptlBrushOrg;
+
+        ProbeForRead(plineattrs, sizeof(*plineattrs), 1);
+        lineattrs = *plineattrs;
+
+        ret = EngStrokePath(pso, ppo, pco, pxo, pbo, &ptlBrushOrg, &lineattrs, mix);
```
Comment (tkreuzer) [inline win32ss/gdi/eng/pathdraw.c]: This is not how it works. The usermode DDI structures cannot simply be used in kernel mode. Instead they need to be translated using UMPD objects.

File: win32ss/gdi/eng/pathdraw.c [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+{
+    BOOL ret = FALSE;
+    POINT ptlBrushOrg;
+
+    _SEH2_TRY
+    {
+        ProbeForRead(pptlBrushOrg, sizeof(*pptlBrushOrg), 1);
+        ptlBrushOrg = *pptlBrushOrg;
+
+        ret = EngFillPath(pso, ppo, pco, pbo, &ptlBrushOrg, mix, flOptions);
```
Comment (tkreuzer) [inline win32ss/gdi/eng/pathdraw.c]: See other comment.

File: win32ss/gdi/eng/pathdraw.c [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+    _SEH2_TRY
+    {
+        ProbeForRead(pptlBrushOrg, sizeof(*pptlBrushOrg), 1);
+        ptlBrushOrg = *pptlBrushOrg;
+
+        ProbeForRead(plineattrs, sizeof(*plineattrs), 1);
+        lineattrs = *plineattrs;
+
+        ret = EngStrokeAndFillPath(pso, ppo, pco, pxo, pboStroke, &lineattrs, pboFill,
```
Comment (tkreuzer) [inline win32ss/gdi/eng/pathdraw.c]: See other comment.

**Comment by jimtabor** [comment]
Have code related to UMPD going in soon.

**Comment by katahiromz** [comment]
I don't know about UMPD objects. Closed.

**Pull request closed without merging**